pvlib/iotools/solaranywhere.py

```diff
@@ -0,0 +1,285 @@
+    weather_request_id = request.json()["WeatherRequestId"]
+
+    # The SolarAnywhere API is asynchronous, hence a second request is
+    # necessary to retrieve the data (WeatherDataResult).
+    start_time = time.time()  # Current time in seconds since the Epoch
+    # Attempt to retrieve results until the max response time has been exceeded
+    while True:
+        results = requests.get(url+'/WeatherDataResult/'+weather_request_id, headers=headers)  # noqa: E501
+        results_json = results.json()
+        if results_json.get('Status') == 'Done':
+            if results_json['WeatherDataResults'][0]['Status'] == 'Failure':
+                raise RuntimeError(results_json['WeatherDataResults'][0]['ErrorMessages'][0]['Message'])  # noqa: E501
+            break
+        elif (time.time()-start_time) > max_response_time:
+            raise TimeoutError('Time exceeded the `max_response_time`.')
+        time.sleep(5)  # Sleep for 5 seconds before each data retrieval attempt
+
+    # Extract time series data
+    data = pd.DataFrame(results_json['WeatherDataResults'][0]['WeatherDataPeriods']['WeatherDataPeriods'])  # noqa: E501
+    # Set index and convert to UTC time
+    data.index = pd.to_datetime(data['ObservationTime'])
+    data.index = data.index.tz_convert('UTC')
+    if map_variables:
+        data = data.rename(columns=VARIABLE_MAP)
+
+    # Parse metadata
+    meta = results_json['WeatherDataResults'][0]['WeatherSourceInformation']
+    meta['time_resolution'] = results_json['WeatherDataResults'][0]['WeatherDataPeriods']['TimeResolution_Minutes']  # noqa: E501
+    # Rename and convert applicable metadata parameters to floats
+    meta['latitude'] = float(meta.pop('Latitude'))
+    meta['longitude'] = float(meta.pop('Longitude'))
+    meta['altitude'] = float(meta.pop('Elevation_Meters'))
+    return data, meta
+
+
+def read_solaranywhere(filename, map_variables=True):
+    """
+    Read a SolarAnywhere formatted file into a pandas DataFrame.
+
+    The SolarAnywhere file format and variables are described in [1]_. Note,
+    the SolarAnywhere file format resembles the TMY3 file format but contains
+    additional variables and meatadata.
+
+    Parameters
+    ----------
+    fbuf: file-like object
+        File-like object containing data to read.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+
+    Returns
+    -------
+    data: pandas.DataFrame
+        Timeseries data from SolarAnywhere. Index is localized to UTC.
+    metadata: dict
+        Metadata available in the file.
+
+    See Also
+    --------
+    pvlib.iotools.get_solaranywhere
+
+    References
+    ----------
+    .. [1] `SolarAnywhere historical data file formats
+       <https://www.solaranywhere.com/support/historical-data/file-formats/>`_
+    """
+    with open(str(filename), 'r', encoding='iso-8859-1') as fbuf:
```

Comment:
Mild preference to expose `encoding` as an optional parameter

Reply:
Also for the ``get_solaranywhere`` function?

Reply:
I don't think `get_solaranywhere` has to worry about encoding problems; network requests shouldn't be subject to users unwittingly re-encoding files.